src/solvers/dgmulti/flux_differencing.jl

```diff
@@ -13,116 +12,169 @@ Computes the flux difference ∑_j A[i, j] * f(u_i, u_j) and accumulates the res
+  fluxdiff_local_threaded = [svectorscopy(zeros(nvars, rd.Nq), Val{nvars}()) for _ in 1:Threads.nthreads()]
+
+  return (; md, Qrst_skew_Tr, sparsity_pattern, invJ = inv.(md.J),
+            u_values, u_face_values, flux_face_values,
+            local_values_threaded, fluxdiff_local_threaded)
+end
 
 function create_cache(mesh::VertexMappedMesh, equations, dg::DGMultiFluxDiff{<:Polynomial}, RealT, uEltype)
 
   rd = dg.basis
   @unpack md = mesh
 
-  Qrst_hybridized, VhP, Ph = StartUpDG.hybridized_SBP_operators(rd)
-  Qrst_skew_Tr = map(A -> -0.5*(A-A'), Qrst_hybridized)
+  Qrst_skew_Tr, VhP, Ph = compute_flux_differencing_SBP_matrices(dg)
+  sparsity_pattern = compute_sparsity_pattern(Qrst_skew_Tr, dg)
 
   nvars = nvariables(equations)
 
+  # temp storage for entropy variables at volume quad points
+  entropy_var_values = allocate_nested_array(uEltype, nvars, (rd.Nq, md.num_elements))
+
   # storage for all quadrature points (concatenated volume / face quadrature points)
   num_quad_points_total = rd.Nq + rd.Nfq
-  entropy_projected_u_values = StructArray{SVector{nvars, uEltype}}(ntuple(_->zeros(num_quad_points_total, md.num_elements), nvars))
-  projected_entropy_var_values = similar(entropy_projected_u_values)
+  entropy_projected_u_values = allocate_nested_array(uEltype, nvars, (num_quad_points_total, md.num_elements))
+  projected_entropy_var_values = allocate_nested_array(uEltype, nvars, (num_quad_points_total, md.num_elements))
 
-  # initialize views into entropy_projected_u_values
+  # initialize temporary storage as views into entropy_projected_u_values
   u_values = view(entropy_projected_u_values, 1:rd.Nq, :)
   u_face_values = view(entropy_projected_u_values, rd.Nq+1:num_quad_points_total, :)
-
-  # temp storage for entropy variables at volume quad points
-  entropy_var_values = StructArray{SVector{nvars, uEltype}}(ntuple(_->zeros(rd.Nq, md.num_elements), nvars))
+  flux_face_values = similar(u_face_values)
 
   # local storage for interface fluxes, rhs, and source
-  flux_face_values = similar(u_face_values)
-  rhs_local_threaded = [StructArray{SVector{nvars, uEltype}}(ntuple(_->zeros(num_quad_points_total), nvars)) for _ in 1:Threads.nthreads()]
-  local_values_threaded = [StructArray{SVector{nvars, uEltype}}(ntuple(_->zeros(rd.Nq), nvars)) for _ in 1:Threads.nthreads()]
+  local_values_threaded = [allocate_nested_array(uEltype, nvars, (rd.Nq,)) for _ in 1:Threads.nthreads()]
 
-  return (; md, Qrst_skew_Tr, VhP, Ph, invJ = inv.(md.J),
+  # We use copy(reinterpret(SVector, vec(x))) to speed up flux differencing.
+  # The result is then transferred to rhs_local_threaded::StructArray{<:SVector} before
+  # projecting it and storing it into `du`.
+  fluxdiff_local_threaded = [svectorscopy(zeros(nvars, num_quad_points_total), Val{nvars}()) for _ in 1:Threads.nthreads()]
+  rhs_local_threaded = [allocate_nested_array(uEltype, nvars, (num_quad_points_total,))  for _ in 1:Threads.nthreads()]
```

Comment:
This comment doesn't seem to fit completely anymore?

Reply:
Can you expand on this? The comment is referring to how `fluxdiff_local_threaded` is used within the solver, and was intended to try to explain why two different array types are used.

Reply:
Updated